### DISCUSSION TITLE
Registering users in Git Proxy Application.Seeking correct process ?

### DISCUSSION THREAD

**Discussion by hemantsharma90** [finos/git-proxy#856] [Q&A]
Hello Team,
We wanted to understand the user management aspects(CRUD Operations of  Users) in the Git-Proxy Application 
viz. 
Registering/Managing Admin Role Users, 
Registering /Managing Contributor Role Users, 
Registering /Managing Reviewer Role Users.

Didn’t found any documented process to do User Registration(maintaining passwords etc) in the gitproxy documentation.

 https://git-proxy.finos.org/docs/. 

From the [source code](https://github.com/finos/git-proxy) of the application , got a fair bit of understanding that we could actually add users by updating [filename: './.data/db/users.db'] , though we reckon there is a much cleaner approach to do the same (which I might be missing while deciphering the source code 😊 ).P.S - I'm not an expert in NodeJS constructs.

Though we can add an approved github/gitlab repository from the ADMIN UI of Git Proxy, and actually can associate registered users as  Reviewers/ Contributors for the respective repositories.

https://github.com/finos/git-proxy/issues/57

However to register a **user** in the first place is something we want to understand.

Regards, 
Hemant Sharma

**Comment by jescalada** [comment]
As mentioned in #868:

GitProxy currently supports adding new users through ActiveDirectory or OIDC login, but no "direct" user creation through email/password. If you'd like to add new users through OIDC, you can follow the setup guide here: https://github.com/finos/git-proxy/pull/906
